niceman/distributions/conda.py

```diff
@@ -92,13 +92,43 @@ def install_packages(self, session=None):
             Environment sub-class instance.
         """
 
-        # TODO: Need to figure out a graceful way to install conda before we can install packages here.
-        # for package in self.provenance['packages']:
-        #     session.add_command(['conda',
-        #                            'install',
-        #                            package['name']])
+        # TODO: Install Conda
+#        print("We got" + str(environment))
+        # Loop through non-root packages, creating the conda-env export
+        # file to import
+        for env in self.environments:
+            if env.path == self.path:
+                continue
+            export_contents = self.create_conda_export(env)
+            print ("EXPORT\n" + export_contents)
+
         return
 
+    @staticmethod
+    def create_conda_export(env):
+        # TODO: The environment name should be discovered on retrace
+        name = os.path.basename(os.path.normpath(env.path))
+        export = "name: %s\n" % name
+        # Collect channels
+        export += "channels:\n"
```

Comment:
Depending on the number of packages that are present, the final export
string may be a result of a good number of string concatenations.  An
alternative to the string concatenation would be to build up a list of
strings (sans the new lines) and then `"\n".join(export_lines)`.

Reply:
I've been spoiled with mutable strings. I'll make the switch.